custom_enumerable.rb

```diff
@@ -0,0 +1,116 @@
+
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity, Style/CaseEquality
+
+
+module Enumerable
+  def my_each
+      i = 0
+      if block_given?
+      while i < self.length 
+          yield(self[i])
+          i += 1
+      end
+      end
+  end
+
+  def my_each_with_index
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns an enumerator if no block is given
```ruby
array.my_each_with_index == Enumerator #true
```